### PR TITLE
refactor(admin): convert NodeView to composition api

### PR DESCRIPTION
**Pull request by cgx**
# Description
Convert NodeView to Vue.js Composition API.

## TODO
- [x] extend `BaseFormGroupChosenOneSearchable` to handle complex lookups
- [x] convert `pfFormDatetime`
- [x] restore validation

### CODE REVIEW

File: html/pfappserver/root/static.alt/src/views/Nodes/_components/FormGroupPid.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { BaseFormGroupChosenOneSearchable, BaseFormGroupChosenOneSearchableProps } from '@/components/new'
```
Comment (satkunas) [inline html/pfappserver/root/static.alt/src/views/Nodes/_components/FormGroupPid.js]: This component should reside at `html/pfappserver/root/static.alt/src/views/Users/_components/FormGroupPid.js`

File: html/pfappserver/root/static.alt/src/views/Nodes/_components/TheForm.vue [hunk 2]
```diff
@@ -0,0 +1,801 @@
...
+    stack: true,
+    tooltip: {
+      followMouse: true
+    }
+  })
+
+  const node = computed(() => $store.state.$_nodes.nodes[id.value])
+  const rolesWithNull = computed(() => {
+    return [
+      { value: null, text: i18n.t('No Role') }, // prepend a null value to roles
```
Comment (satkunas) [inline html/pfappserver/root/static.alt/src/views/Nodes/_components/TheForm.vue]: This option may need to be removed. We no longer use `allowEmpty: false` with vue-multiselect and now allow all inputs to be cleared and handle any required values through validations instead. Therefore an additional empty value isn't needed in some cases to help clear the input. The empty option can be left if it helps clarify the choice(s).

File: html/pfappserver/root/src/views/Nodes/_components/BaseFormGroupRolesWithNull.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        .then(roles => ([
+          { value: null, text: i18n.t('No Role') }, // prepend a null value to roles
+          ...roles.map(role => ({ value: role.category_id, text: `${role.name} - ${role.notes}` }))
+        ]))
+    }
+  }
+}
+
+export default {
+  name: 'base-form-group-roles-with-null',
```
Comment (cgx) [inline html/pfappserver/root/src/views/Nodes/_components/BaseFormGroupRolesWithNull.js]: I propose `BaseFormGroupRolesOptional` instead.

File: html/pfappserver/root/src/views/Nodes/_components/TabDhcpOption82Logs.vue [hunk 4]
```diff
@@ -0,0 +1,62 @@
+<template>
+  <b-tab title="Option82">
+    <template v-slot:title>
+      {{ $t('Option82') }} <b-badge pill v-if="node && node.dhcpoption82 && node.dhcpoption82.length > 0" variant="light" class="ml-1">{{ node.dhcpoption82.length }}</b-badge>
+    </template>
+    <b-table v-if="node && node.dhcpoption82"
+      :items="node.dhcpoption82" :fields="dhcpOption82Fields" :sortBy="dhcpOption82SortBy" :sortDesc="dhcpOption82SortDesc" responsive show-empty sort-icon-left striped>
```
Comment (cgx) [inline html/pfappserver/root/src/views/Nodes/_components/TabDhcpOption82Logs.vue]: Use kebab case in tags.

File: html/pfappserver/root/src/views/Nodes/_components/TheFormCreate.vue [hunk 5]
```diff
@@ -0,0 +1,135 @@
+<template>
+  <b-form @submit.prevent ref="rootRef">
+    <base-form
+      :form="form"
+      :schema="schema"
+      :isLoading="isLoading"
```
Comment (cgx) [inline html/pfappserver/root/src/views/Nodes/_components/TheFormCreate.vue]: Use kebab case in tags.

File: html/pfappserver/root/src/views/Nodes/_components/TheFormCreate.vue [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+        :column-label="$i18n.t('Notes')"
+      />
+      <div class="mt-3">
+        <div class="border-top p-3">
+          <base-form-button-bar
+            isNew
+            :isLoading="isLoading"
+            isSaveable
+            :isValid="isValid"
+            :formRef="rootRef"
```
Comment (cgx) [inline html/pfappserver/root/src/views/Nodes/_components/TheFormCreate.vue]: Use kebab case in tags.

File: html/pfappserver/root/src/views/Nodes/_components/TheFormCreate.vue [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+    isValid,
+    onClose,
+    onCreate,
+    onReset
+  }
+}
+
+// @vue/component
+export default {
+  name: 'the-form-create-single',
```
Comment (cgx) [inline html/pfappserver/root/src/views/Nodes/_components/TheFormCreate.vue]: Doesn't match the filename.

File: html/pfappserver/root/src/views/Nodes/_components/TheFormUpdate.vue [hunk 8]
```diff
@@ -0,0 +1,218 @@
+<template>
+  <b-form @submit.prevent="onSave" ref="rootRef">
+    <base-form
+      :form="form"
+      :schema="schema"
+      :isLoading="isLoading"
```
Comment (cgx) [inline html/pfappserver/root/src/views/Nodes/_components/TheFormUpdate.vue]: Use kebab case in tags.

File: html/pfappserver/root/src/views/Nodes/_components/TheFormUpdate.vue [hunk 9]
```diff
@@ -0,0 +1,218 @@
...
+
+      <div class="mt-3">
+        <div class="border-top pt-3">
+          <form-button-bar class="mr-3"
+            :actionKey="actionKey"
+            :isLoading="isLoading"
+            :isCloneable="false"
+            :isSaveable="true"
+            :isDeletable="isDeletable"
+            :isValid="isValid"
```
Comment (cgx) [inline html/pfappserver/root/src/views/Nodes/_components/TheFormUpdate.vue]: Use kebab case in tags.

File: html/pfappserver/root/src/composables/useInputMultiselect.js [hunk 10]
```diff
@@ -163,40 +163,56 @@ export const useSingleValueLookupOptions = (value, onInput, lookup, options, opt
...
-      }
-    }).then(response => {
-      if (thisCurrentPromise === lastCurrentPromise) { // ignore slow responses
-        const { data: { items = [] } = {} } = response
-        currentValueOptions.value = items.map(item => {
-          const { [fieldName]: _field, [valueName]: _value } = item // unmap lookup field_name/value_name
-          return { [label.value]: _field, [trackBy.value]: _value } // remap option label/trackBy
+
+    if (lookup.value.constructor === Function) { // handle lookup Function (manual)
+      Promise.resolve(lookup.value(value.value, true))
```
Comment (cgx) [inline html/pfappserver/root/src/composables/useInputMultiselect.js]: Magic `true` variable? Add some comment.

File: html/pfappserver/root/src/composables/useInputMultiselect.js [hunk 11]
```diff
@@ -211,46 +227,64 @@ export const useSingleValueLookupOptions = (value, onInput, lookup, options, opt
...
-        const values = query
-          .trim() // trim outside whitespace
-          .split(' ') // separate terms by space
-          .filter(q => q) // ignore multiple spaces
-          .map(query => ({ op: 'or', values: [{ field: fieldName, op: 'contains', value: query }] }))
+    if (lookup.value.constructor === Function) { // handle lookup Function (manual)
+      searchDebouncer({
+        handler: () => {
+          const thisSearchPromise = ++lastSearchPromise
+          Promise.resolve(lookup.value(query, false))
```
Comment (cgx) [inline html/pfappserver/root/src/composables/useInputMultiselect.js]: Magic `false` variable? Add some comment.

**Pull request merged**